base/asyncmap.jl

```diff
@@ -145,4 +145,27 @@ function next(itr::StreamMapIterator, state::StreamMapState)
     return (r, state)
 end
 
-iteratorsize(::Type{StreamMapIterator}) = SizeUnknown()
+iteratorsize(::Type{StreamGenerator}) = SizeUnknown()
+
+
+"""
+    asyncgenerate(f, c...) -> iterator
+
+Apply `@async f` to each element of `c`.
+
+For multiple collection arguments, apply f elementwise.
+
+Results are returned in order as they become available.
+"""
+asyncgenerate(f, c...) = StreamGenerator(f, c...)
```

Comment:
I must say I find it quite confusing that `asyncgenerate` creates a `StreamGenerator` and not an `AsyncGenerator`. Shouldn't `AsyncGenerator` be `AsyncGenerator!`, to parallel `map`/`map!`? Then `StreamGenerator` could be `AsyncGenerator`.

I don't understand why the concept of "stream" is used here: AFAIK we only use this term for `IOStream`, which doesn't seem to be related.